File: client-app/src/app/models/product.ts

import { PurchaseOrder } from "./purchaseOrder";
import { ProductionRecord } from "./productionRecord"; 
import { Material } from "./material";

export interface Product {
    id: string;
    name: string;
    quantity: number;
    description: string;
    image?: any;
    price?: number;
    isProducing : boolean
    choseMaterials: Material[] | null;
    purchaseOrders?: PurchaseOrder[] | null;
    productionRecords?: ProductionRecord[] | null; 
}

File: Domain/Product.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Product
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public long? Quantity { get; set; }
        public int? Price { get; set; }
        public string Description { get; set; }
        public byte[] Image { get; set; }
        public bool IsProducing { get; set; }
        public List<MaterialProduct> MaterialProducts { get; set; }
        public List<PurchaseOrder> PurchaseOrders { get; set; }
        public List<ProductionRecord> ProductionRecords { get; set; }     
    }
}
